sensing/autoware_pointcloud_preprocessor/src/distortion_corrector/distortion_corrector.cpp

```diff
@@ -187,15 +175,139 @@ bool DistortionCorrector<T>::isInputValid(sensor_msgs::msg::PointCloud2 & pointc
+      abs(*next_it_azimuth - *it_azimuth) == 0 ||
+      abs(next_cartesian_rad - current_cartesian_rad) == 0) {
+      RCLCPP_DEBUG_STREAM_THROTTLE(
+        node_->get_logger(), *node_->get_clock(), 10000 /* ms */,
+        "Angle between two points is 0 degrees. Iterate to next point ...");
+      continue;
+    }
+
+    // restrict the angle difference between [-180, 180] (degrees)
+    float azimuth_diff = abs(*next_it_azimuth - *it_azimuth) > autoware::universe_utils::pi
+                           ? abs(*next_it_azimuth - *it_azimuth) - 2 * autoware::universe_utils::pi
+                           : *next_it_azimuth - *it_azimuth;
+    float cartesian_rad_diff =
+      abs(next_cartesian_rad - current_cartesian_rad) > autoware::universe_utils::pi
+        ? abs(next_cartesian_rad - current_cartesian_rad) - 2 * autoware::universe_utils::pi
+        : next_cartesian_rad - current_cartesian_rad;
+
+    float sign = azimuth_diff / cartesian_rad_diff;
+
+    // Check if 'sign' can be adjusted to 1 or -1
+    if (std::abs(sign - 1.0f) <= angle_conversion.sign_threshold) {
+      angle_conversion.sign = 1.0f;
+    } else if (std::abs(sign + 1.0f) <= angle_conversion.sign_threshold) {
+      angle_conversion.sign = -1.0f;
+    } else {
+      RCLCPP_DEBUG_STREAM_THROTTLE(
+        node_->get_logger(), *node_->get_clock(), 10000 /* ms */,
+        "Value of sign is not close to 1 or -1. Iterate to next point ...");
+      continue;
+    }
+
+    float offset_rad = *it_azimuth - sign * current_cartesian_rad;
+    // Check if 'offset_rad' can be adjusted to offset_rad multiple of π/2
+    int multiple_of_90_degrees = std::round(offset_rad / (autoware::universe_utils::pi / 2));
+    if (
+      std::abs(offset_rad - multiple_of_90_degrees * (autoware::universe_utils::pi / 2)) >
+      angle_conversion.offset_rad_threshold) {
+      RCLCPP_DEBUG_STREAM_THROTTLE(
+        node_->get_logger(), *node_->get_clock(), 10000 /* ms */,
+        "Value of offset_rad is not close to multiplication of 90 degrees. Iterate to next point "
+        "...");
+      continue;
+    }
+
+    // Limit the range of offset_rad in [0, 360)
+    multiple_of_90_degrees = (multiple_of_90_degrees % 4 + 4) % 4;
+
+    angle_conversion.offset_rad = multiple_of_90_degrees * (autoware::universe_utils::pi / 2);
+
+    return angle_conversion;
+  }
+  return std::nullopt;
+}
+
+void DistortionCorrectorBase::warn_if_timestamp_is_too_late(
+  bool is_twist_time_stamp_too_late, bool is_imu_time_stamp_too_late)
+{
+  if (is_twist_time_stamp_too_late) {
+    RCLCPP_WARN_STREAM_THROTTLE(
+      node_->get_logger(), *node_->get_clock(), 10000 /* ms */,
+      "Twist time_stamp is too late. Could not interpolate.");
+  }
+
+  if (is_imu_time_stamp_too_late) {
+    RCLCPP_WARN_STREAM_THROTTLE(
+      node_->get_logger(), *node_->get_clock(), 10000 /* ms */,
+      "IMU time_stamp is too late. Could not interpolate.");
+  }
+}
+
+void DistortionCorrectorBase::convert_matrix_to_transform(
```

Comment:
This is a common idiom here, but whenever possible (as in it does not increase the processing time), consider returning instead of modifying a reference

Reply:
True, thanks!
Fixed in https://github.com/autowarefoundation/autoware.universe/pull/8380/commits/b1ffa5e40cef1b554e990e2e0bc24d0660f49fa2